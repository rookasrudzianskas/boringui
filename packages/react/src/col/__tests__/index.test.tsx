import React from "react";
import {mount, render} from "enzyme";

import Col from "../index";

describe("Col", () => {
  it("should render correctly", () => {
    const wrapper = mount(<Col>col</Col>);

    expect(wrapper).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should work with span and offset", () => {
    const wrapper = render(
      <div>
        <Col span={2}>col</Col>
        <Col offset={2} span={2}>
          col
        </Col>
      </div>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should render correctly when nested", () => {
    const wrapper = mount(
      <Col>
        <Col>
          <Col />
          col
        </Col>
      </Col>,
    );

    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should render different components", () => {
    let wrapper = mount(<Col as="p" />);

    expect(wrapper.find("p").length).not.toBe(0);

    wrapper = mount(<Col as="details" />);
    expect(wrapper.find("details").length).not.toBe(0);

    wrapper = mount(<Col as="h1" />);
    expect(wrapper.find("h1").length).not.toBe(0);

    wrapper = mount(<Col as="div" />);
    expect(wrapper.find("div").length).not.toBe(0);
  });
});
