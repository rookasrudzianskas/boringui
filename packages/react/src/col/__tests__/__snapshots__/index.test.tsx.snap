// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col should render correctly 1`] = `ReactWrapper {}`;

exports[`Col should work with span and offset 1`] = `
LoadedCheerio {
  "0": <div>
    <div
      class="nextui-c-cMTZRp nextui-c-cMTZRp-ikSMVmn-css"
    >
      col
    </div>
    <div
      class="nextui-c-cMTZRp nextui-c-cMTZRp-icjwvEK-css"
    >
      col
    </div>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
